src/article.ts

```diff
@@ -182,46 +197,58 @@ const parseElement =
 
 }
 
+type Elements = IBlockElement[] | undefined;
+
 const parseElements =
-    (docParser: DocParser) => (elements: BlockElement[]): Result<string, BodyElement>[] =>
-    elements.map(parseElement(docParser));
+    (docParser: DocParser) => (elements: Elements): Result<string, BodyElement>[] => {
+        if (!elements) {
+            return [new Err('No body elements available')];
+        }
+        return elements.map(parseElement(docParser));
+    }
 
-const parseBlock = (docParser: DocParser) => (block: Block): LiveBlock =>
+const parseBlock = (docParser: DocParser) => (block: IBlock): LiveBlock =>
     ({
         id: block.id,
-        isKeyEvent: block.attributes.keyEvent,
-        title: block.title,
-        firstPublished: block.firstPublishedDate,
-        lastModified: block.lastModifiedDate,
+        isKeyEvent: block?.attributes?.keyEvent ?? false,
+        title: block?.title ?? "",
+        firstPublished: new Date(block?.firstPublishedDate?.iso8601 ?? 0),
+        lastModified: new Date(block?.lastModifiedDate?.iso8601 ?? 0),
         body: parseElements(docParser)(block.elements),
     })
 
-const parseBlocks = (docParser: DocParser) => (blocks: Block[]): LiveBlock[] =>
+const parseBlocks = (docParser: DocParser) => (blocks: IBlock[]): LiveBlock[] =>
     blocks.map(parseBlock(docParser));
 
 const articleFields = (docParser: DocParser, content: Content): ArticleFields =>
     ({
-        pillar: pillarFromString(content.pillarId),
-        headline: content.fields.headline,
-        standfirst: docParser(content.fields.standfirst),
-        byline: content.fields.byline,
-        bylineHtml: fromNullable(content.fields.bylineHtml).map(docParser),
-        publishDate: content.webPublicationDate,
+        pillar: pillarFromString(content?.pillarId),
+        headline: content?.fields?.headline ?? "",
+        standfirst: docParser(content?.fields?.standfirst ?? ""),
+        byline: content?.fields?.byline ?? "",
+        bylineHtml: fromNullable(content?.fields?.bylineHtml).map(docParser),
+        // eslint-disable-next-line
+        // @ts-ignore
+        // CAPI is sending us a string, even though the thrift definition is CapiDateTime
+        publishDate: content?.webPublicationDate,
         mainImage: articleMainImage(content).andThen(parseImage),
         contributors: articleContributors(content),
         series: articleSeries(content),
-        commentable: content.fields.commentable,
+        commentable: content?.fields?.commentable ?? false,
         tags: content.tags,
     })
 
-const articleFieldsWithBody = (docParser: DocParser, content: Content): ArticleFieldsWithBody =>
-    ({
+const articleFieldsWithBody = (docParser: DocParser, content: Content): ArticleFieldsWithBody => {
+    const body = content?.blocks?.body;
+    const elements = body?.length ? body[0]?.elements : undefined;
```

Comment:
That's a bit hard to read. Any chance something like that would work? `body?[0].elements`

Reply:
I don't know if there's a nice way to do an optional lookup on array indices... I tried to hunt around for one but didn't have much luck.

Sadly, I think this will compile fine but error at runtime:

```ts
type Foo = {
    bar: {
        baz: number;
    }
}

const foos: Foo[] = [];

foos[0].bar.baz;
```

Reply:
Perhaps the solution is a simple helper function:

```ts
const getIndex = (idx: number) => <A>(arr: A[]): Option<A> => {
    const value = arr[idx];
    return value !== undefined ? new Some(value) : new None();
  }
```

Reply:
Then I guess it's not worth the effort. It's probably fine as it is